rasa/shared/core/trackers.py

```diff
@@ -459,6 +460,10 @@ def applied_events(self) -> List[Event]:
                 )
                 if event.use_text_for_featurization is None:
                     event.use_text_for_featurization = use_text_for_featurization
+                # update event's entities based on the future event
+                entities = self._define_user_entities(events_as_list[i + 1 :])
+                if event.entities is None:
```

Comment:
not sure about that one. Because most probably `event.entities` is already set with smth from nlu pipeline. I'd say we have to update `entities` dict treating `TEDPolicy` as another extractor. I'm not sure though how the historical `UserUttered` events are persisted. We need to avoid updating entities every time

Maybe we should do it `if event.use_text_for_featurization is None:` but not sure if it is a correct condition, it needs to be tested

Reply:
I updated the check. Currently checking if the entity is not yet inside the entities dict. If not, it is added.